README.md

```diff
@@ -0,0 +1,129 @@
+
+This crate supports insertion of strings already in the Influx Data line
+protocol format. The `influxdb-derive` crate provides convenient serialization
+of Rust structs to the line protocol format.
+
+[InfluxData]: https://www.influxdata.com/
+
+## Examples
+
+To serialize a struct into the Influx Data line protocol format, use the
+`influxdb-derive` crate's macros as shown below with `MyMeasure`.
+
+Then create an instance of `influxdb::AsyncDb` and add instances of your
+struct. Check out the code in the `examples` directory to see how this code
+interacts with futures.
+
+```
+extern crate influxdb;
+#[macro_use]
+extern crate influxdb_derive;
+extern crate tokio_core;
+
+use influxdb::{Measurement, AsyncDb};
+
+// `Measurement` is the trait that AsyncDb needs in order to insert
+#[derive(Measurement)]
+// The default measure name will be the struct name; this optional
+// annotation allows customization of the measure name sent to InfluxDB.
+#[influx(rename = "my_measure")]
+struct MyMeasure {
+    // Specify which struct fields are Influx tags.
+    // Influx tags must always be `String`s.
+    #[influx(tag)]
+    region: String,
+    // Specify which struct fields are Influx fields.
+    // Supported types are integers, floats, strings, and booleans.
+    // The rename annotation works with struct field annotations as well.
+    #[influx(field, rename = "amount")]
+    count: i32,
+    // Struct fields that aren't annotated won't be sent to InfluxDB.
+    other: i32,
+}
+
+fn main() {
+    let mut core = tokio_core::reactor::Core::new()
+        .expect("Unable to create reactor core");
+
+    let async_db = AsyncDb::new(
+        core.handle(),                   // A tokio_core handle
+        "http://localhost:8086/", // URL to InfluxDB
+        "my_database"             // Name of the database in InfluxDB
+    ).expect("Unable to create AsyncDb");
+
+    let batch = vec![
+        MyMeasure { region: String::from("us-east"), count: 3, other: 0 },
+        MyMeasure { region: String::from("us-west"), count: 20, other: 1 },
+    ];
+
+    let insert = async_db.add(&batch); // Returns a Future
+}
+```
+
+## Running the tests
+
+The tests assume that InfluxDB is running and has been configured to accept
+data via UDP.
+
+On macOS, you can install InfluxDB via Homebrew:
+
+```
+brew install influxdb
+```
+
+Then add the UDP configuration by appending the provided
+`tests/influxdb.udp.conf` to the configuration file
+`/usr/local/etc/influxdb.conf` to create a local configuration file:
+
+```
+cat /usr/local/etc/influxdb.conf tests/influxdb.udp.conf > influxdb.conf
+```
+
+And start InfluxDB with that local configuration file:
+
+```
+influxd -config influxdb.conf
+```
+
+On Linux, one way to accomplish the same setup is to follow the steps in
+`.travis.yml`.
+
+Once you have InfluxDB configured and running, run the tests:
+
+```
+cargo test
+```
+
+## Caveats
+
+- Because this library is asynchronous, occasionally tests may fail because of
```

Comment:
Doesn't actually have to do with async; purely sync tests would experience the same problem. It's because InfluxDB receives our request to index data and acknowledges it (ending the HTTP session) before actually doing any indexing.